Mathlib/Topology/CWComplex/Classical/Subcomplex.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2025 Floris van Doorn and Hannah Scholz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Hannah Scholz
+-/
+
+import Mathlib.Topology.CWComplex.Classical.Finite
+import Mathlib.Analysis.Normed.Module.RCLike.Real
+
+/-!
+# Subcomplexes
+
+In this file we discuss subcomplexes of CW complexes.
+The definintion of subcomplexes is in the file `Topology.CWComplex.Classical.Basic`.
+
+## Main results
+* `RelCWComplex.Subcomplex.instRelCWComplex`: a subcomplex of a (relative) CW complex is again a
+  (relative) CW complex.
+
+## References
+* [K. Jänich, *Topology*][Janich1984]
+-/
+
+noncomputable section
+
+open Metric Set
+
+namespace Topology
+
+variable {X : Type*} [t : TopologicalSpace X] {C D : Set X}
+
+/-- A subcomplex is the union of its closed cells and its base. -/
+lemma RelCWComplex.Subcomplex.union_closedCell [T2Space X] [RelCWComplex C D] (E : Subcomplex C) :
+    D ∪ ⋃ (n : ℕ) (j : E.I n), closedCell (C := C) n j = E := by
+  apply subset_antisymm
+  · apply union_subset
+    · rw [← Subcomplex.union (C := C) (D := D) (E := E)]
+      exact subset_union_left
+    · apply iUnion_subset fun n ↦ iUnion_subset fun i ↦ ?_
+      rw [← closure_openCell_eq_closedCell, E.closed.closure_subset_iff, ← E.union]
+      apply subset_union_of_subset_right
+      exact subset_iUnion_of_subset n
+        (subset_iUnion (fun (i : ↑(E.I n)) ↦ openCell (C := C) (D := D) n ↑i) i)
```

Comment:
```suggestion
  · apply union_subset E.base_subset
    apply iUnion₂_subset fun n i ↦ ?_
    rw [← closure_openCell_eq_closedCell, E.closed.closure_subset_iff, ← E.union]
    apply subset_union_of_subset_right
    exact subset_iUnion₂ (s := fun n (i : E.I n) ↦ openCell n (i : cell C n)) n i
```
I think the frequent use of `Subcomplex.union` suggests there should be lemmas saying that if `i∈ E.I n` then `openCell C n i`, `closedCell C n i` and `cellFrontier C n i` are subsets of `E`.